
configurable simple bootflag support

From: Zwane Mwaikambo <zwane@arm.linux.org.uk>
Subject: Re: 2.6.1-rc1-tiny2

Make SBF optional

zwane@montezuma linux-2.6.1-rc1-tiny2 {0:0} size arch/i386/kernel/bootflag.o
   text    data     bss     dec     hex filename
   1645       8       0    1653     675 arch/i386/kernel/bootflag.o

Index: alp-linux/arch/i386/kernel/Makefile
===================================================================
--- alp-linux.orig/arch/i386/kernel/Makefile	2006-07-12 14:18:59.000000000 -0700
+++ alp-linux/arch/i386/kernel/Makefile	2006-07-12 14:22:24.000000000 -0700
@@ -6,7 +6,7 @@
 
 obj-y	:= process.o semaphore.o signal.o entry.o traps.o irq.o \
 		ptrace.o time.o ioport.o ldt.o setup.o i8259.o sys_i386.o \
-		pci-dma.o i386_ksyms.o i387.o dmi_scan.o bootflag.o \
+		pci-dma.o i386_ksyms.o i387.o dmi_scan.o \
 		quirks.o i8237.o topology.o
 
 obj-y				+= cpu/
@@ -39,6 +39,7 @@
 obj-$(CONFIG_EARLY_PRINTK)	+= early_printk.o
 obj-$(CONFIG_KGDB)		+= kgdb.o kgdb-jmp.o
 obj-$(CONFIG_LTT)               += ltt.o
+obj-$(CONFIG_BOOTFLAG)		+= bootflag.o
 
 EXTRA_AFLAGS   := -traditional
 
Index: alp-linux/init/Kconfig
===================================================================
--- alp-linux.orig/init/Kconfig	2006-07-12 14:21:22.000000000 -0700
+++ alp-linux/init/Kconfig	2006-07-12 14:21:30.000000000 -0700
@@ -631,6 +631,13 @@
 	  Select the maximum number of IDE interfaces (0 for default).
           Saves up to 14k.
 
+config BOOTFLAG
+	depends X86
+	default y
+	bool "Enable Simple Bootflag" if EMBEDDED
+	help
+	  This enables support for the Simple Bootflag Specification.
+
 endmenu		# General setup
 
 config CRC32_CALC
